### PR TITLE
Updated basic gtest and float only gpu operations 

### PR DESCRIPTION
**Pull request by cpurc002**
Updated gtests to use test fixtures and gpu operations to handle both float and double data types

### CODE REVIEW

**Comment by yiskylee** [comment]
cpp/include/.gpu_operations.h.swo is committed, please delete it

**Comment by yiskylee** [comment]
cpp/test/matrix_determinant_test.temp is committed, please rename it.

**Comment by cpurc002** [comment]
cpp/test/matrix_determinant_test.temp is the gtest for the gpu determinant that uses test fixtures, but because the cpu version of the determinant hasn't been written I still have to use my old naive version, the cpp/test/matrix_determinant_test.cc to compile. I'll remove it from the pull request for now.

File: cpp/include/gpu_operations.h [hunk 1]
```diff
@@ -44,9 +44,55 @@ namespace Nice {
 template <typename T>
 class GpuOperations {
  public:
+  static cublasStatus_t do_multiply(cublasHandle_t handle, int n,
```
Comment (shidong-ai) [inline cpp/include/gpu_operations.h]: Could you move all these wrapper functions to gpu_util.h and gpu_util.cc? One for declaration and the other for definition. I already put all the wrapper functions I used there. You can take it as a reference.
Besides, the function name you used is also inconsistent with the google code style. I would suggest you change them as well. For example, you can change do_multiply to GpuMatrixScale(...) to make more sense based on the cublas or cusolver api interface called

File: cpp/test/gpu_operations_test/matrix_determinant_test.temp [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    row_ = m;
+    col_ = n;
+
+    // Create matrix
+    a_ = Nice::Matrix<T>::Random(row_, col_);
+    // CPU SVD
+    Nice::CpuOperations<T> cpu_op;
+
+    // Solve in CPU
+    det_ = cpu_op.Determinant(a_);
```
Comment (shidong-ai) [inline cpp/test/gpu_operations_test/matrix_determinant_test.temp]: Given that all member function is static, you can call them simply by Nice::CpuOperations<T>::Determinant(a_);

File: cpp/test/gpu_operations_test/matrix_determinant_test.temp [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    if (a_.rows() != 0 && a_.cols() != 0)
+      return;
+
+    // Set up dimension
+    row_ = m;
+    col_ = n;
+
+    // Create matrix
+    a_ = Nice::Matrix<T>::Random(row_, col_);
+    // CPU SVD
```
Comment (shidong-ai) [inline cpp/test/gpu_operations_test/matrix_determinant_test.temp]: The comments should be modified

**Comment by cpurc002** [comment]
I just saw your comments Shi, I'll get to making those changes tomorrow

File: cpp/include/gpu_util.h [hunk 4]
```diff
@@ -116,6 +116,33 @@ cusolverStatus_t GpuLinearSolver(cusolverDnHandle_t handle,
                                  int ldb,
                                  int *devInfo);
 
+cusolverStatus_t do_get_det_buffer(cusolverDnHandle_t handle,
```
Comment (shidong-ai) [inline cpp/include/gpu_util.h]: Could you change the function name to camel case with Gpu as prefix? For example, do_get_det_buffer should be GpuGetDeterminantBuffer or GpuGetDeterminantWorkspace. So are the other wrapper functions. Many thanks!

**Pull request merged**